internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go

```diff
@@ -120,73 +120,69 @@ func resourceCassandraDatacenter() *pluginsdk.Resource {
+	client := meta.(*clients.Client).Cosmos.ManagedCassandraClient
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	clusterId, _ := parse.CassandraClusterID(d.Get("cassandra_cluster_id").(string))
-	id := parse.NewCassandraDatacenterID(clusterId.SubscriptionId, clusterId.ResourceGroup, clusterId.Name, d.Get("name").(string))
+	clusterId, _ := managedcassandras.ParseCassandraClusterID(d.Get("cassandra_cluster_id").(string))
+	id := managedcassandras.NewDataCenterID(clusterId.SubscriptionId, clusterId.ResourceGroupName, clusterId.CassandraClusterName, d.Get("name").(string))
 
-	existing, err := client.Get(ctx, id.ResourceGroup, id.CassandraClusterName, id.DataCenterName)
+	existing, err := client.CassandraDataCentersGet(ctx, id)
 	if err != nil {
-		if !utils.ResponseWasNotFound(existing.Response) {
+		if !response.WasNotFound(existing.HttpResponse) {
 			return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
 		}
 	}
-	if !utils.ResponseWasNotFound(existing.Response) {
+	if !response.WasNotFound(existing.HttpResponse) {
 		return tf.ImportAsExistsError("azurerm_cosmosdb_cassandra_datacenter", id.ID())
 	}
 
-	body := documentdb.DataCenterResource{
-		Properties: &documentdb.DataCenterResourceProperties{
-			DelegatedSubnetID:  utils.String(d.Get("delegated_management_subnet_id").(string)),
-			NodeCount:          utils.Int32(int32(d.Get("node_count").(int))),
+	body := managedcassandras.DataCenterResource{
+		Properties: &managedcassandras.DataCenterResourceProperties{
+			DelegatedSubnetId:  utils.String(d.Get("delegated_management_subnet_id").(string)),
+			NodeCount:          utils.Int64(int64(d.Get("node_count").(int))),
 			Sku:                utils.String(d.Get("sku_name").(string)),
 			AvailabilityZone:   utils.Bool(d.Get("availability_zones_enabled").(bool)),
-			DiskCapacity:       utils.Int32(int32(d.Get("disk_count").(int))),
+			DiskCapacity:       utils.Int64(int64(d.Get("disk_count").(int))),
 			DiskSku:            utils.String(d.Get("disk_sku").(string)),
 			DataCenterLocation: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
 		},
 	}
 
 	if v, ok := d.GetOk("backup_storage_customer_key_uri"); ok {
-		body.Properties.BackupStorageCustomerKeyURI = utils.String(v.(string))
+		body.Properties.BackupStorageCustomerKeyUri = utils.String(v.(string))
 	}
 
 	if v, ok := d.GetOk("base64_encoded_yaml_fragment"); ok {
 		body.Properties.Base64EncodedCassandraYamlFragment = utils.String(v.(string))
 	}
 
 	if v, ok := d.GetOk("managed_disk_customer_key_uri"); ok {
-		body.Properties.ManagedDiskCustomerKeyURI = utils.String(v.(string))
+		body.Properties.ManagedDiskCustomerKeyUri = utils.String(v.(string))
 	}
 
-	future, err := client.CreateUpdate(ctx, id.ResourceGroup, id.CassandraClusterName, id.DataCenterName, body)
+	err = client.CassandraDataCentersCreateUpdateThenPoll(ctx, id, body)
 	if err != nil {
```

Comment:
FWIW this can become:

```suggestion
	if err = client.CassandraDataCentersCreateUpdateThenPoll(ctx, id, body); err != nil {
```

Reply:
Updated